generic.go

```diff
@@ -60,6 +60,24 @@ func WithImage(image string) CustomizeRequestOption {
 	}
 }
 
+// imageSubstitutor {
+// ImageSubstitutor represents a way to substitute container image names
+type ImageSubstitutor interface {
+	// Description returns the name of the type and a short description of how it modifies the image.
+	// Useful to be printed in logs
+	Description() string
+	Substitute(image string) string
```

Comment:
I have 2 use cases: prefixing and pinning by hash.

For prefixing this works fine.

For pinning by hash, this is awkward because I'd have to write a substitutor that panics when it encounters an unknown image, right? Should I be even thinking about that use case when considering this interface or should I expect images to already be pinned by hash in the testcontainers code itself? I could add a verification in my image substitutor to ensure that images are pinned by hash and panic if they are not, so if something goes wrong upstream and an image is not pinned, it doesn't impact us.

Reply:
@vikstrous2 any feedback from this? I'd like to merge the image substitutor part, probably adding error handling to the substitution part, but want also to understand more on the pinned versions use case you mentioned 🙏

Reply:
Sorry for the delay. Pinning the images mitigates some of the supply chain risk. If your docker hub account is compromised, we wouldn't immediately pick up a malicious image. The code, without being changed, wouldn't change its behavior based on an external dependency.

A recent example of a high profile attack like this is the one that hijacked the circle ci install script: https://circleci.com/blog/jan-4-2023-incident-report/

Re: error handling, yeah, adding an error return parameter is better than having to panic.

Reply:
Ok, will add the error handling there.

Regarding the pinned images, will do it in a separate PR. I now understand the risk in supply chain scenarios. I appreciate for feedback 🙏